components/raftstore-v2/src/batch/store.rs

```diff
@@ -323,12 +401,47 @@ impl<EK: KvEngine, ER: RaftEngine> StoreSystem<EK, ER> {
     }
 
     pub fn shutdown(&mut self) {
+        if self.workers.is_none() {
+            return;
+        }
+        let mut workers = self.workers.take().unwrap();
+
         self.apply_system.shutdown();
         self.system.shutdown();
+
+        workers.store_writers.shutdown();
+        workers.log_fetch_worker.stop();
+    }
+}
+
+#[derive(Clone)]
+pub struct StoreRouter<EK: KvEngine, ER: RaftEngine> {
+    router: BatchRouter<PeerFsm<EK, ER>, StoreFsm>,
+    logger: Logger,
```

Comment:
Can we use a global logger instead of passing a logger everywhere?

Reply:
Using a logger explicitly is better for following two reasons:
- Custom KVs can be added, for example, no need to write "peer_id" => peer_id, "region_id" => region_id everywhere.
- Slightly faster access, because we don't need to check for global loggers.